src/hookintrprtr.h

```diff
@@ -0,0 +1,178 @@
+/****************************************************************************
+**
+*W  hookintrprtr.h                    GAP source              Chris Jefferson
+**
+**
+*Y  Copyright (C) 2017 The GAP Group
+**
+**  This file contains functions related to hooking the interpreter.
+**  This code was abstracted out from the line-by-line profiling, for other
+**  uses.
+**
+*/
+
+#ifndef GAP_HOOKINTRPRTR_H
+#define GAP_HOOKINTRPRTR_H
+
+#include "exprs.h"
+
+void InstallEvalBoolFunc( Int, Obj(*)(Expr));
+void InstallEvalExprFunc( Int, Obj(*)(Expr));
+void InstallExecStatFunc( Int, UInt(*)(Stat));
+void InstallPrintStatFunc(Int, void(*)(Stat));
+void InstallPrintExprFunc(Int, void(*)(Expr));
+
+
+/****************************************************************************
+**
+** Store the true values of each function we wrap for profiling. These always
+** store the correct values and are never changed.
+**
+** These are provided for interpreter hooks to call the original methods.
+*/
```

Comment:
I know that you are just following the example of the rest of the kernel, but: I am actually not a big fan of duplicating all those documentating comments. If we have two copies, we need to always remember to edit them in both places, and invariably they'll get out of sync anyway.

Of course that leaves the decision which one should be kept: the one in the header, or the one in the source? :-).

Anyway, I am not expecting you to change this right now, just wanted to mention it.